GoogleForCreators/web-stories-wp
# Dashboard: Make stories list view header sticky to keep in view on scroll

## Summary
Updates the stories list view to have a sticky table heading so that when stories are scrolled through in list view the headers stay visible to avoid confusion if focus is on them without disrupting where the user is on the view by moving the page.

## Relevant Technical Choices
- Uses `squishContentHeight` and `DASHBOARD_TOP_MARGIN` to give the table heading its `top:` value with `position: sticky`. The top value is passed to the stickyheader component as a prop so that it's not a fixed number. 
- Because our dashboard layout uses a provider to "squish" the space of the header on scroll I had to tweak the way the header sees it's space a little bit so that the values stay constant regardless of if there's a banner in there or not.
  - Basically, this boils down to wrapping the banner and setting height to the wrapper so that padding can be taken from the space provided (thus the bump from 161 to 171 since now that 10px margin top is coming from that value). 
  - And then, no more padding for the pageHeader top - that space is now decided on by the heading itself. 
- Adjusted the z-index so table header is covering the images that scroll past it. 

## To-do

- Combine this with Dillon's update for #4483 

## User-facing changes

- Table header of stories list view should now become "sticky" when you scroll through stories making it stay visible. 

## Testing Instructions

Scroll through stories in list view, see that table heads stay in view and are interactive. Test with and without telemetry banner. 

![sticky scroll with banner](https://user-images.githubusercontent.com/10720454/93137685-b88b0c80-f692-11ea-8f7c-9f1c5fb18528.gif)
![sticky scroll](https://user-images.githubusercontent.com/10720454/93137695-bc1e9380-f692-11ea-9ade-b1088fa9b5d3.gif)


---



addresses #4420
